hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTableRegionLocatorImpl.java

```diff
@@ -61,8 +62,17 @@ public CompletableFuture<List<HRegionLocation>> getAllRegionLocations() {
         return conn.registry.getMetaRegionLocations()
           .thenApply(locs -> Arrays.asList(locs.getRegionLocations()));
       }
-      return ClientMetaTableAccessor
+      CompletableFuture<List<HRegionLocation>> future = ClientMetaTableAccessor
         .getTableHRegionLocations(conn.getTable(TableName.META_TABLE_NAME), tableName);
+      addListener(future, (locs, error) -> {
+        if (error != null) {
+          future.completeExceptionally(error);
```

Comment:
when i say re-complete I mean i think it may not be necessary to call completeExceptionally here or complete below. Instead just return if error != null

Reply:
Yes, you are right. It is unnecessary to call re-complete. I'll remove it. Thanks.